frontend/src/container/CreateAlertV2/EvaluationSettings/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,155 @@
+import './TimeInput.scss';
+
+import { Input } from 'antd';
+import React, { useEffect, useState } from 'react';
+
+export interface TimeInputProps {
+	value?: string; // Format: "HH:MM:SS"
+	onChange?: (value: string) => void;
+	disabled?: boolean;
+	className?: string;
+}
+
+function TimeInput({
+	value = '00:00:00',
+	onChange,
+	disabled = false,
+	className = '',
+}: TimeInputProps): JSX.Element {
+	const [hours, setHours] = useState('00');
+	const [minutes, setMinutes] = useState('00');
+	const [seconds, setSeconds] = useState('00');
+
+	// Parse initial value
+	useEffect(() => {
+		if (value) {
+			const timeParts = value.split(':');
+			if (timeParts.length === 3) {
+				setHours(timeParts[0].padStart(2, '0'));
+				setMinutes(timeParts[1].padStart(2, '0'));
+				setSeconds(timeParts[2].padStart(2, '0'));
+			}
+		}
+	}, [value]);
+
+	// Format time value
+	const formatTimeValue = (h: string, m: string, s: string): string =>
+		`${h.padStart(2, '0')}:${m.padStart(2, '0')}:${s.padStart(2, '0')}`;
+
+	// Handle input change
+	const handleTimeChange = (
+		newHours: string,
+		newMinutes: string,
+		newSeconds: string,
+	): void => {
+		const formattedValue = formatTimeValue(newHours, newMinutes, newSeconds);
+		onChange?.(formattedValue);
+	};
+
+	// Handle hours change
+	const handleHoursChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newHours = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setHours(newHours);
+		handleTimeChange(newHours, minutes, seconds);
+	};
+
+	// Handle minutes change
+	const handleMinutesChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newMinutes = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setMinutes(newMinutes);
+		handleTimeChange(hours, newMinutes, seconds);
+	};
+
+	// Handle seconds change
+	const handleSecondsChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newSeconds = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setSeconds(newSeconds);
+		handleTimeChange(hours, minutes, newSeconds);
+	};
```

Comment:
These are allowing invalid valid for h, m, and s? As in, one can give 80th minute and 94th second